### PR TITLE
.editorconfig: Add file to speed up editor configuration

### PR DESCRIPTION
**Pull request by c3r1c3**
…irements

For more details: http://editorconfig.org/
VS2017 supports this natively.
VS2015 and earlier can use a plug-in.
Most other popular IDEs support this natively or via a plug-in.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# EditorConfig is awesome: http://EditorConfig.org
+# Since OBS follow the Linux Kernel Conventions
```
Comment (RytoEX) [inline .editorconfig]: Spelling/Grammar: "follow" should be "follows"

These lines (1-4, or 2-4) could be wrapped much wider, as they have about 30 characters to spare before hitting the 80 character limit.

Also, I'm not sure about using "Linux Kernel Conventions"?  As far as I know, the phrase describing the coding style of OBS is one of these:

- kernel normal form (linux variant)
- KNF (Linux)
- Linux kernel coding style

See [CONTRIBUTING.rst](https://github.com/jp9000/obs-studio/blob/master/CONTRIBUTING.rst#coding-guidelines).

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+indent_style = tab
+indent_size = 8
```
Comment (RytoEX) [inline .editorconfig]: Missing new-line at the end of this file.

**Comment by c3r1c3** [comment]
Thanks for the review.

Missing Newline: Added.
Change VS2015 "plug-in" to "extension": Done.
Made the VS2015 line actually be something close to proper English: Done. ;-)
Fix line lengths: Done.
Change/shorten message commit title: Done.

Did everything but change 'plug-in' to 'plugin'. While I personally don't care if someone uses one form or the other, it bugs me to not put in the hyphen.

**Comment by RytoEX** [comment]
I understand.  For years, I used "e-mail" instead of "email".  I don't know why GitHub doesn't recognize that my newline comment is solved, but eh.

**Comment by notr1ch** [comment]
I'd recommend adding this for compatibility with 3rd party code that's part of obs-outputs:

```
[plugins/obs-outputs/librtmp/*.{cpp,c,h}]
indent_style = space
```

**Comment by c3r1c3** [comment]
Thanks notr1ch! Added.

**Pull request merged**